crates/net/src/builder.rs

```diff
@@ -180,14 +179,18 @@ impl NetworkDriverBuilder {
             )?
             .with_behaviour(|_| behaviour)?
             .build();
-        let addr = self.socket.take().ok_or_else(|| eyre::eyre!("socket address not set"))?;
-        let addr = NetworkAddress::try_from(addr)?;
-        let swarm_addr = Multiaddr::from(addr);
-        let gossip = GossipDriver::new(swarm, swarm_addr, handler);
+        let socket = self.socket.take().ok_or_else(|| eyre::eyre!("socket address not set"))?;
```

Comment:
we should be able to do this here

```suggestion
        let socket = self.socket.take().ok_or(eyre::eyre!("socket address not set"))?;
```